lib/SIL/Parser/SILParsedLocalValueMap.h

```diff
@@ -0,0 +1,71 @@
+//===--- SILParser.h - SIL Undef Value Representation -----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SIL_PARSER_H
+#define SWIFT_SIL_PARSER_H
+
+#include "swift/AST/DeclContext.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Parse/Parser.h"
+#include "swift/SIL/SILInstruction.h"
+
+namespace swift {
+namespace syntax {
+
+//===----------------------------------------------------------------------===//
+// SILParser local value container
+//===----------------------------------------------------------------------===//
+
+struct UnresolvedValueName {
+  StringRef Name;
+  SourceLoc NameLoc;
+
+  bool isUndef() const { return Name == "undef"; }
+};
+
+/// Holds SIL local values and forward references.
+class SILParsedLocalValueMap {
+  SILModule &module;
+
+  llvm::StringMap<SILValue> localValues;
+  llvm::StringMap<SourceLoc> forwardRefs;
```

Comment:
Can you expand out forwardRefs -> forwardReferences and maybe add a comment on how it is used?

Reply:
Also, I noticed above that there was a note about non-determinism when iterating over forward refs. Why don't we just nip that in this PR as well by changing forwardRefs to be a MapVector.

Reply:
Good call. Done.